src/Arcus.EventGrid.Tests.Integration/Fixture/EventGridTopicEndpoint.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Threading.Tasks;
+using Arcus.EventGrid.Contracts;
+using Arcus.EventGrid.Publishing;
+using Arcus.EventGrid.Publishing.Interfaces;
+using Arcus.EventGrid.Testing.Infrastructure.Hosts.ServiceBus;
+using Arcus.EventGrid.Tests.Integration.Logging;
+using GuardNet;
+using Xunit.Abstractions;
+
+namespace Arcus.EventGrid.Tests.Integration.Fixture 
+{
+    /// <summary>
+    /// Represents an EventGrid topic endpoint that can be interacted with by publishing events.
+    /// </summary>
+    public class EventGridTopicEndpoint
+    {
+        private readonly EventSchema _eventSchema;
+        private readonly TestConfig _configuration;
+
+        private EventGridTopicEndpoint(
+            EventSchema eventSchema,
+            ServiceBusEventConsumerHost serviceBusEventConsumerHost,
+            TestConfig config)
+        {
+            Guard.NotNull(serviceBusEventConsumerHost, nameof(serviceBusEventConsumerHost));
+            Guard.NotNull(config, nameof(config));
+
+            _eventSchema = eventSchema;
+            _configuration = config;
+            ServiceBusEventConsumerHost = serviceBusEventConsumerHost;
+        }
+
+        /// <summary>
+        /// Gets the consumer host on the current topic endpoint.
+        /// </summary>
+        public ServiceBusEventConsumerHost ServiceBusEventConsumerHost { get; }
+
+        /// <summary>
+        /// Builds a <see cref="IEventGridPublisher"/> implementation that interacts with this endpoint.
+        /// </summary>
+        public IEventGridPublisher BuildPublisher()
+        {
+            string topicEndpoint = _configuration.GetEventGridTopicEndpoint(_eventSchema);
+            string endpointKey = _configuration.GetEventGridEndpointKey(_eventSchema);
+            
+            IEventGridPublisher publisher =
+                EventGridPublisherBuilder
+                    .ForTopic(topicEndpoint)
+                    .UsingAuthenticationKey(endpointKey)
+                    .Build();
+
+            return publisher;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="EventGridTopicEndpoint"/> implementation that uses CloudEvent's as input event schema.
+        /// </summary>
+        /// <param name="testOutput">The test logger to write diagnostic messages during the availability of the endpoint.</param>
+        public static async Task<EventGridTopicEndpoint> CreateForCloudEventAsync(ITestOutputHelper testOutput)
```

Comment:
Aren't we mixing creating the endpoint publisher and consuming the events in one class?

Reply:
LGTM